Arriba/Arriba/Model/Query/SelectQuery.cs

```diff
@@ -210,25 +202,49 @@ public void Correct(ICorrector corrector)
             this.Where = corrector.Correct(this.Where);
         }
 
+        private SelectContext SafeGetContext(Partition p)
```

Comment:
Slightly tidier option. You could change Prepare() to take ITable instead of Table, and only do anything if this.Context is null. Then you could call it from both OnBeforeQuery(table) and Compute(Partition). That way at least there's only one copy of the "set a default OrderByColumn" and "expand * to all columns" logic and it runs for both tests and real use.